### PR TITLE
10314 Fix Conversation impressions not increasing

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-10314

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
@dominicmarmionswrve @shanetmoore Could you have a look?

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java [hunk 1]
```diff
@@ -340,4 +340,12 @@ protected boolean checkCampaignLimits(String event, Date now, Map<Integer, Strin
 
         return true;
     }
+
+    /**
+     * Used by sublcasses to inform that the campaign was displayed.
+     */
+    public void messageWasShownToUser() {
+        incrementImpressions();
+        setMessageMinDelayThrottle();
```
Comment (shanetmoore) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java]: Are the delay throttles for conversations and IAM shared?

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 2]
```diff
@@ -235,6 +236,9 @@ public void onMessage(final SwrveConversation conversation) {
                                 intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                                 intent.putExtra("conversation", conversation);
                                 ctx.startActivity(intent);
+
+                                // Report that the message was shown to users
+                                conversation.getCampaign().messageWasShownToUser();
```
Comment (shanetmoore) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: We send impression events at the moment for conversations on a per page basis (I think)

How will this extra impression event affect reporting?

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java (hunk 1 above)
Comment (Sergio-Mira) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java]: Yeah

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveBase.java (hunk 2 above)
Comment (Sergio-Mira) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: That method does not send an impression event. It increments impressions internally (a counter used to check later on) and sets the min message delay

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveCampaign.java [hunk 3]
```diff
@@ -174,8 +174,7 @@ protected SwrveMessage createMessage(SwrveBase<?, ?> controller, SwrveCampaign s
      * Notify that a message was shown to the user.
      */
     public void messageWasShownToUser() {
```
Comment (dominicmarmionswrve) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveCampaign.java]: add @Override annotation?

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java [hunk 4]
```diff
@@ -340,4 +340,12 @@ protected boolean checkCampaignLimits(String event, Date now, Map<Integer, Strin
 
         return true;
     }
+
+    /**
+     * Used by sublcasses to inform that the campaign was displayed.
+     */
+    public void messageWasShownToUser() {
+        incrementImpressions();
```
Comment (dominicmarmionswrve) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java]: make incrementImpressions() private now?

**Comment by dominicmarmionswrve** [comment]
:rocket:

File: SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java (hunk 4 above)
Comment (Sergio-Mira) [inline SwrveCommonSDK/src/main/java/com/swrve/sdk/messaging/SwrveBaseCampaign.java]: Done

**Pull request merged**